packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -826,6 +827,13 @@ - (void)startVideoRecordingWithCompletion:(void (^)(FlutterError *_Nullable))com
                                      details:nil]);
       return;
     }
+    // do not call startWriting in didOutputSampleBuffer to prevent state in which
+    // stopVideoRecordingWithCompletion does not send completion when _isRecording is
+    // YES but _videoWriter.status is AVAssetWriterStatusUnknown and video lag at start
+    // https://github.com/flutter/flutter/issues/132016
+    // https://github.com/flutter/flutter/issues/151319
+    [_videoWriter startWriting];
```

Comment:
do you intend to remove `startWriting` from `didOutputSampleBuffer`?

Reply:
yes